### PR TITLE
feat: Improve thread metrics

### PR DESCRIPTION
**Pull request by gabriellsh**
## Proposed changes (including videos or screenshots)

Updated the thread metrics. Changes include elements positioning, titles and a new avatar "list" of the users following the thread.
image
image
image

## Issue(s)

[CORE-542](https://rocketchat.atlassian.net/browse/CORE-542)
## Further comments

Depends on #33214

[CORE-542]: https://rocketchat.atlassian.net/browse/CORE-542?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .changeset/lazy-avocados-whisper.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"@rocket.chat/meteor": minor
+"@rocket.chat/i18n": minor
+---
+
+Improved thread metrics to feature user avatars, among other improvements.
```
Comment (KevLehman) [inline .changeset/lazy-avocados-whisper.md:6]: Feel we can explain what the "other improvements" are in the changeset unless they're def not visible by users.

File: apps/meteor/client/components/message/content/ThreadMetrics.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+						.withUserPreference('clockMode', 1)
+						.withSetting('Message_TimeFormat', 'LT')
+						.withTranslations(...mockedTranslations)
+						.buildWithRouter(navigateCallback),
+				},
+			);
+
+			const followButton = screen.getByTitle('Following');
+			expect(followButton).toBeVisible();
+			// userEvent.click(followButton);
```
Comment (KevLehman) [inline apps/meteor/client/components/message/content/ThreadMetrics.spec.tsx:115]:
```suggestion
```

File: packages/rest-typings/src/v1/chat.ts [hunk 3]
```diff
@@ -885,7 +893,7 @@ export type ChatEndpoints = {
 	};
 	'/v1/chat.getThreadsList': {
 		GET: (params: ChatGetThreadsList) => {
-			threads: IMessage[];
+			threads: IThreadMainMessage[];
```
Comment (matheusbsilva137) [inline packages/rest-typings/src/v1/chat.ts:896]: Can we ensure that this type is indeed being returned by `chat.getThreadsList`? I see we're still returning `IMessage[]` there

**Review by matheusbsilva137 — CHANGES_REQUESTED** [review]

File: apps/meteor/client/components/message/content/ThreadMetricsParticipants.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import ThreadMetricAvatar from './ThreadMetricAvatar';
+
+type ThreadMetricsParticipantsProps = {
+	participants: Array<string>;
+};
+
+const ThreadMetricsParticipants = ({ participants }: ThreadMetricsParticipantsProps): ReactElement => {
+	const t = useTranslation();
+
+	const participantsLabel = participants.length - 2 > 0 ? `+${participants.length - 2}` : undefined;
```
Comment (MarcosSpessatto) [inline apps/meteor/client/components/message/content/ThreadMetricsParticipants.tsx:15]: What does `participants.length - 2` mean? Wouldn't it be clearer to have a named variable with the meaning (business speaking) of this?

**Review by KevLehman — COMMENTED** [review]
Also, a small Pr desc featuring the changes would be nice too 🤗

File: apps/meteor/app/api/server/v1/chat.ts [hunk 5]
```diff
@@ -533,7 +533,7 @@ API.v1.addRoute(
 			};
 
 			const threadQuery = { ...query, ...typeThread, rid: room._id, tcount: { $exists: true } };
-			const { cursor, totalCount } = await Messages.findPaginated(threadQuery, {
+			const { cursor, totalCount } = await Messages.findPaginated<IThreadMainMessage>(threadQuery, {
```
Comment (KevLehman) [inline apps/meteor/app/api/server/v1/chat.ts:536]: Problem is still `normalizeMessagesForUser` is returning a `Promise<IMessage>`. Could be that the function has to be modified to accept an optional type param too, so you can make it return IThreadMainMessage.

Not sure if the change is worth it 🤔 considering typescript won't complain about it. wdyt?

File: apps/meteor/client/views/room/contextualBar/Threads/components/ThreadListMessage.tsx [hunk 6]
```diff
@@ -19,13 +17,13 @@ type ThreadListMessageProps = {
 	username: IMessage['u']['username'];
 	name?: IMessage['u']['name'];
 	ts: IMessage['ts'];
-	replies: ReactNode;
-	participants: ReactNode;
-	handleFollowButton: MouseEventHandler;
+	replies: number;
+	participants: Array<string> | undefined;
```
Comment (dougfabris) [inline apps/meteor/client/views/room/contextualBar/Threads/components/ThreadListMessage.tsx:21]:
```suggestion
	participants: string[] | undefined;
```

**Review by dougfabris — CHANGES_REQUESTED** [review]

File: apps/meteor/app/lib/server/functions/sendMessage.ts [hunk 7]
```diff
@@ -283,6 +283,8 @@ export const sendMessage = async function (user: any, message: any, room: any, u
 		message._id = insertedId;
 	}
 
+	console.log('Inserted message', message);
```
Comment (MarcosSpessatto) [inline apps/meteor/app/lib/server/functions/sendMessage.ts:286]:
```suggestion
```

**Review by MarcosSpessatto — CHANGES_REQUESTED** [review]

File: apps/meteor/app/lib/server/functions/sendMessage.ts (hunk 7 above)
Comment (gabriellsh) [reply apps/meteor/app/lib/server/functions/sendMessage.ts:286]: I did notice that my test log was missing from on my other branch 🤣

**Review by dougfabris — APPROVED** [review]
Approving on behalf of frontend team!

**Review by dougfabris — APPROVED** [review]

**Review by KevLehman — APPROVED** [review]
BE looking good (for me)

**Review by matheusbsilva137 — APPROVED** [review]

**Review by MarcosSpessatto — DISMISSED** [review]

**Review by KevLehman — DISMISSED** [review]

**Review by MarcosSpessatto — APPROVED** [review]

**Review by dougfabris — APPROVED** [review]

**Comment by matheusbsilva137** [comment]
I believe this PR introduced a lint issue on develop :grimacing:

**Pull request merged**